Sources/SwiftSyntax/SourceLocation.swift

```diff
@@ -43,7 +43,7 @@ struct ComputedLocation: Codable, CustomDebugStringConvertible {
 }
 
 /// Represents a source location in a Swift file.
-public struct SourceLocation: Codable, CustomDebugStringConvertible {
+public struct SourceLocation: Hashable, Codable, CustomDebugStringConvertible {
```

Comment:
Just to be clear: Should two `SourceLocation`s be different if one has a `ComputedLocation` and the other does not? After all, they still refer to the same location in the source code.

I’m not 100% sure on this, but I’m leaning towards making the identity of `SourceLocation` be purely based on the value of `offset`. What do you think, @MaxDesiatov? 

CC: @akyrtzi

Reply:
In my opinion they shouldn't be equal since `line`, `column`, and `file` will return different results for them. The one without a `ComputedLocation` could refer to a different file altogether, as `file` is inferred from `compLoc`.

Reply:
That’s a good point. I didn’t consider `file`. With that in mind, I